src/app/components/chat-box/chat-box.component.ts

```diff
@@ -39,24 +32,33 @@ export class ChatBoxComponent {
   TotalMessages: any;
   NoUser: boolean = true;
   ChatBox: boolean = false;
-
+  isGroup: boolean = false;
   constructor(
     private chatservice: ChatService,
     private location: Location,
     private route: Router,
     private loader: NgxSpinnerService,
-  ) {}
+    public dialog: MatDialog
+  ) {
+  
+  }
   ngOnInit() {
     this.loader.show();
     this.chatservice.UserLoginData.subscribe((res: any) => {
       this.currentUser = res;
     });
     this.chatservice.sendSocketData({
-      data: this.currentUser._id,
+      data: {userId : this.currentUser._id},
       key: 'newUser',
     });
-    this.chatservice.getSocketData('newUser').subscribe((res) => {
-      this.UserListData = res;
+    this.chatservice.getSocketData('newUser').subscribe(({ userPayload, userId, opponentPayload, opponentId}) => {
+     if(this.currentUser._id == userId){
+       this.UserListData = this.currentUser.isAdmin ? userPayload : this.getOnlyAdmins(userPayload);
+     }else if(this.currentUser._id == opponentId){
+      this.UserListData =this.currentUser.isAdmin ? opponentPayload : this.getOnlyAdmins(userPayload);
```

Comment:
change userPayload to adminPayload in the 58th line

Reply:
done